File: src/ios/MinimalPlugin.m
#import <Cordova/CDVPlugin.h>

@interface MinimalPlugin : CDVPlugin
- (void)echo:(CDVInvokedUrlCommand*)command;
@end

@implementation MinimalPlugin

- (void)echo:(CDVInvokedUrlCommand*)command {
    NSString* message = [command.arguments objectAtIndex:0];
    CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:message];
    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}

@end

File: src/ios/MinimalPlugin.swift
import Foundation
import Cordova

@objc(MinimalPlugin) class MinimalPlugin: CDVPlugin {
    @objc(echo:)
    func echo(command: CDVInvokedUrlCommand) {
        let message = command.arguments.first as? String ?? "No message"
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK,
                                           messageAs: "Echo: \(message)")
        self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
    }
}
